mongodbatlas/provider.go

```diff
@@ -125,95 +125,25 @@ func decodeStateID(stateID string) map[string]string {
 }
 
 func valRegion(reg interface{}, opt ...string) (string, error) {
-
-	regions := []string{
```

Comment:
I would actually prefer if we went back to the hardcoded list of supported regions, perhaps we could move it out to a seperate file like `regions.go` for easier maintenance - this gives us a reference for what regions are supported, and easier troubleshooting, rather than letting everything through.

Reply:
We can simply add `ME_SOUTH_1` to the list

Reply:
I thought that in the beginning, but if on your side makes a lot of changes removing and adding we need to maintain that file all the time, instead we can let this validation works on MongoDB server-side to show the correct information for the user. I don't have any problem to get back the list, but I think we need another opinion about this, What do you think @marinsalinas?

Reply:
Can’t you create function that call all available regions from mongodb api?

Reply:
That is a really nice idea 👍 , but I don't know if it exists an entry point to retrieve all the available regions, let me take a look to verify it 👀 .

Reply:
you might have one problem, some instances types are not available in some regions, you need to do probably some error handling or logic on your side.

Reply:
Sorry guys to interrupt your discussion. I don't know every detail about how should be designed providers but if Atlas API validates all requests and doesn't start creation if the payload is invalid why a provider should validate payload on its own side? You can easily send a request and handle the error if it appeared.

Sorry again, but this feature might crucial for my project and should be fixed as soon as possible.